### PR TITLE
test(quantic): adding support for rga components in the insight use case

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-5201](https://coveord.atlassian.net/browse/SFINT-5201)

PS: This PR needs to be merged after [SFINT-5199](https://github.com/coveo/ui-kit/pull/3489)

- After creating the Headless Generated Answer controllers for the insight use case the Quantic Generated Answer component directly started supporting this use case.
- Adapted our Cypress E2E tests to cover the generated answer feature in the insight use case.

#### Demo of the Quantic Generated Answer component in the insight use case:

https://github.com/coveo/ui-kit/assets/86681870/4b80a9c8-2d0e-4f4e-8ff8-abb04156633b

[SFINT-5201]: https://coveord.atlassian.net/browse/SFINT-5201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[SFINT-5199]: https://coveord.atlassian.net/browse/SFINT-5199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticGeneratedAnswer/exampleQuanticGeneratedAnswer.html [hunk 1]
```diff
@@ -7,10 +7,10 @@
       </c-configurator>
     </div>
 
-    <c-quantic-search-interface slot="preview" engine-id={engineId}>
+    <c-example-use-case slot="preview" use-case={config.useCase} engine-id={engineId}>
```
Comment (SimonMilord) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticGeneratedAnswer/exampleQuanticGeneratedAnswer.html:10]: nice touch!

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 2]
```diff
@@ -198,432 +164,500 @@ describe('quantic-generated-answer', () => {
...
-              Actions.clickRetry();
-              cy.wait(InterceptAliases.Search);
-              Expect.logRetryGeneratedAnswer(streamId);
+          it('should not show the component', () => {
+            Expect.displayGeneratedAnswerCard(false);
+          });
+        });
+
+        describe('Retryable error', () => {
+          [500, 429].forEach((errorCode) => {
```
Comment (SimonMilord) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:642]: are those the only 2 possible error codes we can receive?

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 3]
```diff
@@ -198,432 +164,500 @@ describe('quantic-generated-answer', () => {
...
+            describe(`${errorCode} error`, () => {
+              beforeEach(() => {
+                mockSearchWithGeneratedAnswer(streamId, param.useCase);
+                mockStreamError(streamId, errorCode);
+                visitGeneratedAnswer({useCase: param.useCase});
+              });
+
+              it('should retry the stream 3 times then offer a retry button', () => {
+                for (let times = 0; times < 3; times++) {
```
Comment (SimonMilord) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:651]:
```suggestion
                for (let retries = 0; retries < 3; retries++) {
```

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 4]
```diff
@@ -52,144 +58,104 @@ const genQaMessageTypePayload = {
 };
 
 describe('quantic-generated-answer', () => {
+  beforeEach(() => {
+    cy.clock(0, ['Date']);
```
Comment (SimonMilord) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:62]: why do we need this? To wait for the answer to be generated fully?

**Review by SimonMilord — APPROVED** [review]
few questions, but otherwise looks fine to me

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts (hunk 2 above)
Comment (mmitiche) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:642]: Yep.

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts (hunk 4 above)
Comment (mmitiche) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:62]: No, this is for the test where we will hover over a citation, in order to log analytics event we need to hover over the citation for one second or more.

So to win time and avoid making our test to wait, I'm playing with the clock using this Cypress functionality, so now no need to wait we will fast forward time at our convenience.

Comment (SimonMilord) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:62]: ok cool!

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 5]
```diff
@@ -52,144 +58,104 @@ const genQaMessageTypePayload = {
...
   const pageUrl = 's/quantic-generated-answer';
 
   function visitGeneratedAnswer(options: Partial<GeneratedAnswerOptions> = {}) {
     interceptSearch();
     cy.visit(pageUrl);
     configure(options);
+    if (options.useCase === useCaseEnum.insight) {
+      InsightInterfaceExpect.isInitialized();
+      performSearch();
+    }
```
Comment (erocheleau) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:80]: I think I would prefer this to be either in it's own method, or in the "beforeEach()" instead.
It's not really "visitGeneratedAnswer" it's a distinct behavior

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts (hunk 2 above)
Comment (erocheleau) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:642]: Put them in a variable please :)

**Review by erocheleau — APPROVED** [review]
I would perhaps rename this PR to specify this is only writing the tests?

And the linked jira seems weird? It specifies to add GenAI to the insight panel, but this is testing genAI with the insight use-case only no?

**Pull request merged**